runtime-sdk/modules/contracts/src/lib.rs

```diff
@@ -563,6 +565,31 @@ impl<Cfg: Config> Module<Cfg> {
         Ok(types::CallResult(data))
     }
 
+    #[handler(call = "contracts.ChangeUpgradePolicy")]
+    pub fn tx_change_upgrade_policy<C: TxContext>(
+        ctx: &mut C,
+        body: types::ChangeUpgradePolicy,
+    ) -> Result<(), Error> {
+        let params = Self::params(ctx.runtime_state());
+
+        <C::Runtime as Runtime>::Core::use_tx_gas(ctx, params.gas_costs.tx_change_upgrade_policy)?;
+
+        if !ctx.should_execute_contracts() {
```

Comment:
This does not execute contracts so this check can be replaced with just a check for "check tx mode".